### PR TITLE
Promote  kubelet resource metric e2e test to Conformance

### PR DESCRIPTION
**Pull request by richabanker**
#### What type of PR is this?
/area conformance

#### What this PR does / why we need it:
Follow up on https://github.com/kubernetes/kubernetes/pull/116897#discussion_r1267434370 

#### Which issue(s) this PR fixes:
Fixes [#727 ](https://github.com/kubernetes/enhancements/issues/727)

Testgrid Link:
[Testgrid](https://testgrid.k8s.io/sig-instrumentation-tests#gci-gce&include-filter-by-regex=Metrics.should.grab.all.metrics.from.kubelet)

@kubernetes/sig-architecture-pr-reviews @kubernetes/sig-instrumentation-pr-reviews @kubernetes/cncf-conformance-wg

### CODE REVIEW

File: test/conformance/testdata/conformance.yaml [hunk 1]
```diff
@@ -3346,4 +3353,10 @@
     was configured with a subpath.
   release: v1.12
   file: test/e2e/storage/subpath.go
-
+- testname: Kubelet resource metrics
+  codename: '[sig-instrumentation] Metrics should grab all metrics from kubelet''s
+    /metrics/resource endpoint [Conformance]'
```
Comment (rexagod) [inline test/conformance/testdata/conformance.yaml:3358]:
```suggestion
  codename: '[sig-instrumentation] Metrics should grab all metrics from kubelet /metrics/resource
    endpoint [Conformance]'
```
Nit.

Comment (Richabanker) [reply test/conformance/testdata/conformance.yaml]: Done, thanks!

**Review by logicalhan — APPROVED** [review]
/lgtm
/approve

**Comment by johnbelamaric** [comment]
Did you follow the process here: https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/conformance-tests.md#promoting-tests-to-conformance ? I am wondering because I am not sure how the nit got there since that file is populated automatically?

Do you have a link showing the non-flakiness of the test?

**Comment by Richabanker** [comment]
> Did you follow the process here: https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/conformance-tests.md#promoting-tests-to-conformance ? I am wondering because I am not sure how the nit got there since that file is populated automatically?
> 
> Do you have a link showing the non-flakiness of the test?

Oh I was not aware of the formal process regarding [promoting-tests-to-conformance](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/conformance-tests.md#promoting-tests-to-conformance), thanks for sharing that. I have updated the PR description as mentioned in the link. The nit, was a naive mistake on my end, I had first made the changes to the [conformance.yaml](https://github.com/kubernetes/kubernetes/pull/120473/files#diff-96054f175317e3d3dac5c6f71ba8273ee991e7aba2f16a352fb400ebdb89895f) manually, but later used `hack/update-conformance-yaml.sh` to update it.

**Comment by johnbelamaric** [comment]
Hi @Richabanker we can't make this part of conformance. Conformance is focused primarily on user-portability. Requiring that a distribution support a specific Kubelet API endpoint is too prescriptive as to how distributions may actually implement things, and also does not allow managed hosting providers to limit user access to those APIs.

See the second bullet in https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/conformance-tests.md#conformance-test-requirements

**Pull request closed without merging**